bundles/admin/admin-announcements/view/AnnouncementsHelper.js

```diff
@@ -0,0 +1,83 @@
+export const announcementsHelper = () => {
+    const getAdminAnnouncements = (handler) => {
+        if (typeof handler !== 'function') {
+            return;
+        }
+
+        jQuery.ajax({
+            type: 'GET',
+            dataType: 'json',
+            url: Oskari.urls.getRoute('Announcements'),
+            success: function (pResp) {
+                handler(null, pResp);
+            },
+            error: function (jqXHR, textStatus) {
+                handler('Error', []);
+            }
+        });
+    };
+
+    const saveAnnouncement = (data, handler) => {
+        if (typeof handler !== 'function') {
+            return;
+        }
+        jQuery.ajax({
+            type: 'POST',
+            dataType: 'json',
+            contentType: 'application/json; charset=UTF-8',
+            data: JSON.stringify(data),
+            url: Oskari.urls.getRoute('Announcements'),
+            success: function (pResp) {
+                handler(null, pResp);
+            },
+            error: function (jqXHR, textStatus) {
+                handler('Error', []);
+            }
+        });
+    };
+
+    const updateAnnouncement = (data, handler) => {
+        if (typeof handler !== 'function') {
+            return;
+        }
+        jQuery.ajax({
+            type: 'PUT',
+            dataType: 'json',
+            contentType: 'application/json; charset=UTF-8',
+            data: JSON.stringify(data),
+            url: Oskari.urls.getRoute('Announcements'),
+            success: function (pResp) {
+                handler(null, pResp);
+            },
+            error: function (jqXHR, textStatus) {
+                handler('Error', []);
+            }
+        });
+    };
+
+    const deleteAnnouncement = (data, handler) => {
+        if (typeof handler !== 'function') {
+            return;
+        }
+
+        jQuery.ajax({
+            type: 'DELETE',
+            dataType: 'json',
+            data: JSON.stringify(data),
+            url: Oskari.urls.getRoute('Announcements'),
```

Comment:
This makes the id in data-object to be sent as parameter in the url so you can use the getRequiredParam() on the server to validate it:
```suggestion
            url: Oskari.urls.getRoute('Announcements', { id: data.id }),
```

You could also modify the caller to just sent the id as param and not the data object to make it easier to understand/more maintainable.